base/reflection.jl

```diff
@@ -246,6 +251,8 @@ fieldname(t::Type{<:Tuple}, i::Integer) =
 
 Get a tuple with the names of the fields of a `DataType`.
 
+For a tuple type, each name is an `Int` (the index of the field), otherwise it's a `Symbol`.
```

Comment:
I would spell out the colloquialism "it's" as "it is".

Perhaps also have the "common" case first? Like this:
```suggestion
The field names are of type `Symbol`, except for tuple types, where they are
of type `Int` (the index of the field).
```